### ISSUE TITLE
[documentSymbol] Stream / delay results

### ISSUE THREAD

**Issue by ejgallego** [ejgallego/coq-lsp#124]
Currently `textDocument/documentSymbols` will only be served when the document checking is complete.

This is far from ideal, so we can either postpone it to a "document completed queue" or even better, stream the results.

Result streaming would require a modification to the request serving architecture.

**Comment by ejgallego** [comment]
When should fleche give back control to the handler so it can process pending requests is a bit of an open question, maybe we'd like to do that in a time-based heuristic? Still unsure, for now let's let the client drive that with the interruption.

**Comment by Alizter** [comment]
@ejgallego I agree with using interruption for the time being. We can experiment with another technique in the future.
